packages/gatsby/src/utils/webpack.config.js

```diff
@@ -451,47 +451,62 @@ module.exports = async (program, directory, suppliedStage) => {
   }
 
   if (stage === `build-html` || stage === `develop-html`) {
+    // Packages we want to externalize to save some build time
+    // https://github.com/gatsbyjs/gatsby/pull/14208#pullrequestreview-240178728
     const externalList = [
-      // match `lodash` and `lodash/foo`
-      // but not things like `lodash-es`
-      `lodash`,
-      /^lodash\//,
-      `react`,
-      /^react-dom\//,
-      `pify`,
-      `@reach/router`,
       `@reach/router/lib/history`,
+      `@reach/router`,
       `common-tags`,
-      `path`,
-      `semver`,
-      `react-helmet`,
-      `minimatch`,
-      `fs`,
       /^core-js\//,
-      `es6-promise`,
       `crypto`,
-      `zlib`,
-      `http`,
-      `https`,
       `debug`,
+      `fs`,
+      `https`,
+      `http`,
+      `lodash`,
+      `path`,
+      `semver`,
+      /^lodash\//,
+      `zlib`,
     ]
 
+    // Packages we want to externalize because meant to be user-provided
+    const userExternalList = [
+      `es6-promise`,
+      `minimatch`,
+      `pify`,
+      `react-helmet`,
+      `react`,
+      /^react-dom\//,
+    ]
+
+    const checkItem = (item, request) => {
+      if (typeof item === `string` && item === request) {
```

Comment:
should we refactor to something like?
`return typeof item === `string` && item === request || item instanceof RegExp && item.test(request)`

Reply:
I think the current writing might be more readable (I find complex conditions easier to parse when they're split in semantical blocks)

Reply:
make sense :)